src/routes/LostItemsContainer.js

```diff
@@ -1,23 +1,196 @@
+  NoPermissionMessage,
+} from '../views/LostItems';
+import {
+  ACTUAL_COST_RECORD_FIELD_NAME,
+  ACTUAL_COST_RECORD_FIELD_PATH,
+  PAGE_AMOUNT,
+  SEARCH_FIELDS,
+} from '../views/LostItems/constants';
 
-const LostItemsContainer = ({ stripes }) => {
-  const hasPermission = stripes.hasPerm('ui-users.lost-items.requiring-actual-cost');
+function buildFilterConfig(filters) {
+  const customFilterConfig = [];
+  const parsedFilters = parseFilters(filters);
 
-  return (
-    !hasPermission && <NoPermissionMessage content={<FormattedMessage id="ui-users.lostItems.message.noAccessToActualCostPage" />} />
-  );
-};
+  Object.keys(parsedFilters).forEach(name => {
+    if (name.match('customFields')) {
+      customFilterConfig.push(
+        {
+          name,
+          cql: name.split('-').join('.'),
+          values: [],
+          operator: '=',
+        },
+      );
+    }
+  });
 
-LostItemsContainer.propTypes = {
-  stripes: PropTypes.shape({
-    hasPerm: PropTypes.func.isRequired,
-  }).isRequired,
-};
+  return customFilterConfig;
+}
 
-export default withStripes(LostItemsContainer);
+const filterConfig = [
+  {
+    name: ACTUAL_COST_RECORD_FIELD_PATH[ACTUAL_COST_RECORD_FIELD_NAME.LOSS_TYPE],
+    cql: ACTUAL_COST_RECORD_FIELD_PATH[ACTUAL_COST_RECORD_FIELD_NAME.LOSS_TYPE],
+    values: [],
+  },
+];
+
+function buildQuery(queryParams, pathComponents, resourceData, logger, props) {
+  const customFilterConfig = buildFilterConfig(queryParams.filters);
+
+  return makeQueryFunction(
+    'cql.allRecords=1',
+    SEARCH_FIELDS.map(index => `${index}="%{query.query}*"`).join(' or '),
+    {},
+    [...filterConfig, ...customFilterConfig],
+    2,
+  )(queryParams, pathComponents, resourceData, logger, props);
+}
+
+class UserSearchContainer extends React.Component {
+  static manifest = Object.freeze({
+    query: {
+      initialValue: {},
+    },
+    resultCount: {
+      initialValue: PAGE_AMOUNT,
+    },
+    resultOffset: {
+      initialValue: 0,
+    },
+    records: {
+      type: 'okapi',
+      records: 'actualCostRecords',
+      resultOffset: '%{resultOffset}',
+      perRequest: PAGE_AMOUNT,
+      path: 'actual-cost-record-storage/actual-cost-records',
+      GET: {
+        params: {
+          query: buildQuery,
+        },
+      },
+    },
+  });
+
+  static propTypes = {
+    location: PropTypes.object,
+    history: PropTypes.object,
+    mutator: PropTypes.shape({
+      records: PropTypes.shape({
+        POST: PropTypes.func.isRequired,
+      }).isRequired,
+      query: PropTypes.shape({
+        update: PropTypes.func.isRequired,
+        replace: PropTypes.func.isRequired,
+      }).isRequired,
+      resultOffset: PropTypes.shape({
+        replace: PropTypes.func.isRequired,
+      }).isRequired,
+    }).isRequired,
+    stripes: PropTypes.shape({
+      logger: PropTypes.object.isRequired
+    }).isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.logger = props.stripes.logger;
```

Comment:
What is the reason to create `this.logger` if we can directly use `this.props.stripes.logger` on 118 line?

Reply:
Fixed